Building on FreeBSD fails

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

```
error[E0425]: cannot find function `is_library_file` in this scope
   --> src/coprocessor_v2/plugin_registry.rs:133:24
    |
133 |                     if is_library_file(file) {
    |                        ^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `is_library_file` in this scope
   --> src/coprocessor_v2/plugin_registry.rs:248:20
    |
248 |                 if is_library_file(&file) {
    |                    ^^^^^^^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
```

### What version of TiKV are you using?
<!-- You can run `tikv-server --version` -->

### What operating system and CPU are you using?
<!-- If you're using Linux, you can run `cat /proc/cpuinfo` -->
FreeBSD 14.0

### Steps to reproduce
<!-- If possible, provide a recipe for reproducing the error. A complete runnable program is good. -->

### What did you expect?

### What did happened?

--- a/src/coprocessor_v2/plugin_registry.rs
+++ b/src/coprocessor_v2/plugin_registry.rs
@@ -458,7 +458,7 @@ impl CoprocessorPlugin for LoadedPlugin {
     }
 }
 
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
 fn is_library_file<P: AsRef<Path>>(path: P) -> bool {
     path.as_ref().extension() == Some(OsStr::new("so"))
 }

